core/src/test/java/com/google/cloud/sql/core/InternalConnectorRegistryTest.java

```diff
@@ -367,6 +368,107 @@ public void testGetApplicationNameFailsAfterInitialization() {
         () -> InternalConnectorRegistry.setApplicationName("sample-app"));
   }
 
+  @Test
+  public void registerConnection() throws IOException, InterruptedException {
+    FakeSslServer sslServer = new FakeSslServer();
+    int port = sslServer.start(PUBLIC_IP);
+    ConnectionInfoRepositoryFactory factory =
+        new StubConnectionInfoRepositoryFactory(fakeSuccessHttpTransport(Duration.ofSeconds(0)));
+    InternalConnectorRegistry coreSocketFactory =
+        new InternalConnectorRegistry(
+            clientKeyPair, factory, credentialFactory, port, TEST_MAX_REFRESH_MS, defaultExecutor);
+    // Register a ConnectionConfig named "my-connection"
+    ConnectionConfig real =
+        new ConnectionConfig.Builder()
+            .withCloudSqlInstance("myProject:myRegion:myInstance")
+            .withIpTypes("PRIMARY")
+            .build();
+    coreSocketFactory.register("my-connection", real);
+    // Attempt to connect using the cloudSqlNamedConnection connection property
+    Properties connProps = new Properties();
+    connProps.setProperty(ConnectionConfig.CLOUD_SQL_NAMED_CONNECTION_PROPERTY, "my-connection");
+    ConnectionConfig namedConfig = ConnectionConfig.fromConnectionProperties(connProps);
+    Socket socket = coreSocketFactory.connect(namedConfig);
+    // Assert that the socket opens correctly.
+    assertThat(readLine(socket)).isEqualTo(SERVER_MESSAGE);
```

Comment:
How does this assertion verify the registered config was used? Could we simplify this assertion to make that point more clear?

Reply:
Sorry I still don't understand -- how do we know the registered connector was used and not simply an unnamed connector?

Reply:
After the most recent change to ConnectorConfig (from ConnectionConfig), this test doesn't actually test anything anymore. 

I've updated this to clearly test that the named connector is used.